docs/content/en/docs/getting-started/metrics/_index.md

```diff
@@ -4,3 +4,315 @@ description: Learn how Keptn metrics enhances your deployment
+
+```yaml
+kind: KeptnMetricsProvider
+metadata:
+  name: dev-dynatrace
+  namespace: simplenode-dev
+spec:
+  type: dynatrace
+  targetServer: "https://hci34192.live.dynatrace.com"
+  secretKeyRef
+    name: dynatrace
+    key: DT_TOKEN
+...
+```
+
+### Define KeptnMetric information
+
+The [KeptnMetric](../../yaml-crd-ref/metric.md) resource
+defines the information you want to gather,
+specified as a query for the particular observability platform
+you are using.
+You can define any type of metric from any data source.
+
+In our example, we define two bits of information to retrieve:
+
+* Number of CPUs, derived from the `dev-prometheus` data platform
+* `availability` SLO, derived from the `dev-dynatrace` data platform
+
+Each of these are configured to fetch data every 10 seconds
+but you could configure a different `fetchIntervalSeconds` value
+for each metric.
+
+The
+[keptn-metric.yaml](https://github.com/keptn-sandbox/klt-on-k3s-with-argocd/blob/main/simplenode-dev/keptn-metric.yaml)
+file for our example looks like:
+
+```yaml
+apiVersion: metrics.keptn.sh/v1alpha2
+kind: Keptnmetric
+metadata:
+  name: available-cpus
+  namespace: simplenode-dev
+spec:
+  provider:
+    name: dev-prometheus
+  query: "sum(kube_node_status_cvapacity{resources`cpu`})"
+  fetchIntervalSeconds: 10
+...
+apiVersion: metrics.keptn.sh/v1alpha2
+kind: Keptnmetric
+metadata:
+  name: availability-slo
+  namespace: simplenode-dev
+spec:
+  provider:
+    name: dev-dynatrace
+  query: "func:slo.availability_simplenodeservice"
+  fetchIntervalSeconds: 10
+...
+```
+
+Note the following:
+
+* Populate one YAML file per metric
+  then apply all of them.
+* Each metric is assigned a unique `name`.
+* The value of the `spec.provider.name` field
+  must correspond to the name assigned in a
+  the `metadata.name` field of a `KeptnMetricsProvider` resource.
+* Information is fetched in on a continuous basis
+at a rate specified by the value of the `spec.fetchIntervalSeconds` field.
+
+### View available metrics
+
+Use the following command to view
+the metrics that are configured in your cluster.
+This example displays the two metrics we configured above:
+
+TODO: Is the syntax of the output changed
+to include the `name` of the provider (`dev-prometheus` or `dev-dynatrace`?
+
+```shell
+get KeptnMetrics -A
+```
+
+```shell
+NAMESPACE       NAME              PROVIDER   QUERY
+simplenode-dev  availability-slo  dynatrace  func:slo.availability_simplenodeservice
+simplenode-dev  available-cpus    prometheus sum(kube_node_status_capacity{resource=`coy})
+```
+
+## Run the metrics
+
+TODO: Do I need to start and stop anything to start gathering metrics
+or could I theoretically just put these pieces into my cluster
+and would it start gathering metrics that I could then view?
+
+## Observing the metrics
+
+TODO: Do we want to say anything about running these metrics,
+viewing the results, perhaps from CLI and from Grafana?
```

Comment:
Here I would show how we can retrieve the metrics through CRDs and through the Kubernetes Metric API.

From CRDs the syntax is `kubectl get keptnmetrics.metrics.keptn.sh -n <namespace> <metric-name>`. Here's an example with the output for the `available-cpus`

```bash
$ kubectl get keptnmetrics.metrics.keptn.sh -n simplenode-dev available-cpus
NAME             PROVIDER     QUERY                                           VALUE
cpu-throttling   my-provider  sum(kube_node_status_capacity{resource=`coy})   6.000
```

For the kubernetes APIs, the syntax is the following `kubectl get --raw "/apis/custom.metrics.k8s.io/v1beta2/namespaces/<namespace>/keptnmetrics.metrics.sh/<metric-name>/<metric-name>"`. Here's an example with the output for the `available-cpus`

```bash
$ kubectl get --raw "/apis/custom.metrics.k8s.io/v1beta2/namespaces/simplenode-dev/keptnmetrics.metrics.sh/available-cpus/available-cpus"
{
  "kind": "MetricValueList",
  "apiVersion": "custom.metrics.k8s.io/v1beta2",
  "metadata": {},
  "items": [
    {
      "describedObject": {
        "kind": "KeptnMetric",
        "namespace": "simplenode-dev",
        "name": "available-cpus",
        "apiVersion": "metrics.keptn.sh/v1alpha2"
      },
      "metric": {
        "name": "available-cpus",
        "selector": {}
      },
      "timestamp": "2023-05-11T08:05:36Z",
      "value": "6"
    }
  ]
}
```

Reply:
Wonderful!  It's in!